test/libsolidity/StandardCompiler.cpp

```diff
@@ -154,6 +154,30 @@ BOOST_AUTO_TEST_CASE(no_sources)
 	BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
 }
 
+BOOST_AUTO_TEST_CASE(no_sources_empty_object)
+{
+	char const* input = R"(
+	{
+		"language": "Solidity",
+		"sources": {}
+	}
+	)";
+	Json::Value result = compile(input);
+	BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
+}
+
+BOOST_AUTO_TEST_CASE(no_sources_empty_array)
+{
+	char const* input = R"(
+	{
+		"language": "Solidity",
+		"sources": []
```

Comment:
This should result in an error stating it is an invalid input.

Reply:
I think its nicely reflected with `No input sources specified.` - not sufficient?

Reply:
Looking at the other tests it seems we do not have any more granular reporting, so it should be fine for the moment. Can revisit in the future.

Reply:
Wait a second, `[ "aaa", "bbb" ]` will be accepted by the empty-ness check but what will `sources.getMemberNames()` do?

Reply:
```
{
	"errors" : 
	[
		{
			"component" : "general",
			"formattedMessage" : "JSON logic exception: in Json::Value::getMemberNames(), value must be objectValue",
			"message" : "JSON logic exception: in Json::Value::getMemberNames(), value must be objectValue",
			"severity" : "error",
			"type" : "InternalCompilerError"
		}
	]
}
```
I guess we need another check for that,  another PR?

Reply:
I'd just do it in this one.

Reply:
Make sense.